fdbrpc/sim2.actor.cpp

```diff
@@ -2305,6 +2311,60 @@ class Sim2 final : public ISimulator, public INetworkConnections {
 		dieWhenTerminate = false;
 	}
 
+	int forkSearch(const char* context = "") override {
+#if defined(_WIN32)
+		printf("start fork...");
+		return 0;
+#elif defined(__unixish__)
+		if (forkSearchDepth > 0) { // now only allow 1
+			return 0;
+		}
+		std::string parentGroup = getTraceLogGroup();
+		pid_t processId;
+		int status;
+		for (int i = 0; i < forkSearchTimes; ++i) {
+			if ((processId = fork()) == 0) { // child process
+				++forkSearchDepth;
+				int pid = getpid();
+				int new_seed = platform::getRandomSeed(); // non-deterministic seed
+				// TODO: do something to record the seed in order to replay
+
+				std::string childLogGroup = parentGroup + "/" + std::to_string(pid); // format to still be finalized
```

Comment:
do we want to add the new seed into the string?

Reply:
I personally think this would make the loggroup field too long. I think the new seed can be checked in the following trace event